lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2677,6 +2678,12 @@ alreadyVoted walletState =
     Dlgs.isVoting . view #delegations
         <$> readDBVar walletState
 
+haveWeVoted
+    :: DBLayer IO s
+    -> IO Bool
+haveWeVoted DBLayer{..} = do
+    atomically (alreadyVoted walletState)
```

Comment:
When rebasing on #4477, please use `alreadyVoted` directly in the pure functions `joinStakePoolDelegationAction` and `Cardano.Wallet.Delegation`; this function will become obsolete.